### PR TITLE
Improved 404 and 500 error pages

### LINKED ISSUE #1138: Better styling and more verbose information on 404 and 500 errors
* 500 and 404 errors are a regular part of developing with Evidence, but the feedback you receive in the browser is not particularly clear. 
* Error pages in the browser are displaying 500 or 404 codes, but would be more useful to our users if they were in plain english. 
* Style the error message, and include a copy/pasteable output of the full error

### PR DESCRIPTION
**Pull request by ItsMeBrianD**
### Description

image
image

Resolves #1138

### Checklist

- [x] For UI or styling changes, I have added a screenshot or gif showing before & after
- [x] I have added a [changeset](https://github.com/evidence-dev/evidence/blob/main/CONTRIBUTING.md#adding-a-changeset)

### CODE REVIEW

File: sites/example-project/src/hooks.client.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
+/** @param {Error | unknown} e  */
+const transformError = (e) => {
+	if (!(e instanceof Error)) {
+		console.log({ e });
```
Comment (csjh) [inline sites/example-project/src/hooks.client.js:4]: Is this intentionally left in?

File: sites/example-project/src/hooks.server.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
+/** @param {Error | unknown} e  */
+const transformError = (e) => {
+	if (!(e instanceof Error)) {
+		console.log({ e });
```
Comment (csjh) [inline sites/example-project/src/hooks.server.js:4]: ^

**Review by csjh — COMMENTED** [review]
Are the new `example-project` files copied over to `template` properly? They might have to be added to the `@evidence-dev/evidence` build script

**Comment by mcrascal** [comment]
Couple styling items: 
* The page /test-charts/ _can't be found in the project._
* Error code (404/ 500) moved out of the h1 and into the error details. H1 for the page should just be page not found / application error.

**Review by mcrascal — APPROVED** [review]
Styling LGTM (with a couple edits above).
